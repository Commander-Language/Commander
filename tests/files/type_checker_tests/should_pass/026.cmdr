//Test bindings type check
x: fn (int, int) -> int = fn (a: int, b: int): int -> a + b;
y = x;
