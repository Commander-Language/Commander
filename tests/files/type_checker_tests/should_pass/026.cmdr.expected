(PrgmNode
	(ExprStmtNode (BinOpExprNode (BindingNode x (FunctionTypeNode (TypesNode (IntTypeNode INT) (IntTypeNode INT)) (IntTypeNode INT) ((INT, INT) -> INT))) SET (LambdaExprNode (BindingsNode (BindingNode a (IntTypeNode INT)) (BindingNode b (IntTypeNode INT))) (IntTypeNode INT) (ReturnStmtNode (BinOpExprNode (LValueExprNode (VarExprNode a INT) INT) ADD (LValueExprNode (VarExprNode b INT) INT) INT)) ((INT, INT) -> INT)) ((INT, INT) -> INT)))
	(ExprStmtNode (BinOpExprNode (BindingNode y) SET (LValueExprNode (VarExprNode x ((INT, INT) -> INT)) ((INT, INT) -> INT)) ((INT, INT) -> INT))))
